Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/FieldAccessorFunctors.H

```diff
@@ -0,0 +1,40 @@
+#ifndef WARPX_FIELD_ACCESSOR_FUNCTORS_H
+#define WARPX_FIELD_ACCESSOR_FUNCTORS_H
+#include "WarpX.H"
+/**
+ * \brief Functor that returns the division of the source m_field Array4 value
+          by macroparameter, m_parameter value at the respective (i,j,k,ncomp).
+ */
+struct FieldAccessorMacroscopic
+{
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    FieldAccessorMacroscopic ( amrex::Array4<amrex::Real const> const a_field,
+                               amrex::Array4<amrex::Real const> const a_parameter )
+        : m_field(a_field), m_parameter(a_parameter) {}
+
+    /**
+     * \brief return field value at (i,j,k,ncomp) scaled by (1/m_parameters(i,j,k,ncomp))
+     * \param[in] i      index along x of the Array4, m_field and m_parameter.
+     * \param[in] j      index along y of the Array4, m_field and m_parameter.
+     * \param[in] k      index along z of the Array4, m_field and m_parameter.
+     * \param[in] ncomp  index along fourth component of the Array4, containing field-data
+                         to be returned after diving with zero-th component
+                         of m_paramter.
+     *
+     * \return           m_field/m_paramter at (i,j,k,ncomp)
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    amrex::Real operator() (int const i, int const j,
+                            int const k, int const ncomp) const noexcept
+    {
+        return ( m_field(i, j, k, ncomp)  /  m_parameter(i, j, k, 0) ) ;
```

Comment:
just so that it might be helpful in the future, if sometime we are interested in looking at material properties in tensor-form, `m_parameter(i, j, k, 0)` could also be `m_parameter(i, j, k, ncomp)`, but no worries for now.

Reply:
Yes @jackieyao0114. But, just ncomp would not ensure that both field and m_parameter would have the same component. we may then need to separate ncomp_field and ncomp_material_property. So I enforced this to be zero for now to distinguish that the field and m_parameter may not have the same ncomp.